joblib/memory.py

```diff
@@ -541,6 +544,29 @@ def _cached_call(self, args, kwargs, shelving=False):
 
         return (out, args_id, metadata)
 
+    @property
+    def func_code_info(self):
+        # 3-tuple property containing: the function source code, source file,
+        # and first line of the code inside the source file
+        if hasattr(self.func, '__code__'):
+            if self._func_code_id is None:
+                self._func_code_id = id(self.func.__code__)
+            elif id(self.func.__code__) != self._func_code_id:
+                # Be robust to dynamic reassignments of self.func.__code__
+                self._func_code_info = None
+
+        if self._func_code_info is None:
+            # Cache the source code of self.func . Provided that get_func_code
+            # (which should be called once on self) gets called in the process
+            # in which self.func was defined, this caching mechanism prevents
+            # undesired cache clearing when the cached function is called in
+            # an environement where the introspection utilitiees get_func_code
+            # relies on do not work (typicially, in joblib child processes).
+            # See #1035 for  more info
+            # TODO (pierreglaser): do the same with get_func_name?
```

Comment:
That's a good question. It does not seem to be required to fix #1035 though (I tried manually with the reproducer in an interactive ipython session).

Reply:
I also tried to stop and restart the ipython session and the cache survived.